File: src/Java/Screw/Api/ScrewApiClient.java
import java.io.IOException;
import java.net.URI;
import java.net.http.HttpClient;
import java.net.http.HttpHeaders;
import java.net.http.HttpRequest;
import java.net.http.HttpResponse;
import java.nio.charset.StandardCharsets;
import java.util.HashMap;
import java.util.Map;
import com.fasterxml.jackson.databind.ObjectMapper;

public class ScrewApiClient {
    private final HttpClient httpClient;
    private final String apiUrl = "https://api.screwltd.com/";

    public ScrewApiClient() {
        httpClient = HttpClient.newBuilder()
                .baseUrl(URI.create(apiUrl))
                .version(HttpClient.Version.HTTP_1_1)
                .build();
    }

    public String authenticate(String login, String password) throws IOException, InterruptedException {
        Map<String, String> authData = new HashMap<>();
        authData.put("login", login);
        authData.put("password", password);

        String authJson = new ObjectMapper().writeValueAsString(authData);

        String response = postRequest("auth", authJson);
        if (response != null && !response.isEmpty()) {
            Map<String, String> jsonResponse = new ObjectMapper().readValue(response, Map.class);
            return jsonResponse.containsKey("access_key") ? jsonResponse.get("access_key") : null;
        }

        return null;
    }

    public boolean checkAuthValid(String accessKey) {
        return accessKey != null && !accessKey.isEmpty();
    }

    public String createAppKey(String accessKey) throws IOException, InterruptedException {
        String response = postRequest("create_app", "", accessKey);
        if (response != null && !response.isEmpty()) {
            Map<String, String> jsonResponse = new ObjectMapper().readValue(response, Map.class);
            return jsonResponse.containsKey("app_key") ? jsonResponse.get("app_key") : null;
        } else {
            System.out.println("Failed to create app key. Response is empty.");
            return null;
        }
    }

    public boolean authenticateApp(String appKey, String accessKey) throws IOException, InterruptedException {
        String queryParams = "?app_key=" + appKey;
        String response = postRequest("auth_app" + queryParams, "", accessKey);

        if (response != null && !response.isEmpty()) {
            Map<String, Object> jsonResponse = new ObjectMapper().readValue(response, Map.class);
            return jsonResponse.containsKey("result") && (boolean) jsonResponse.get("result");
        }
        return false;
    }

    public int checkAppKeyActivations(String appKey) throws IOException, InterruptedException {
        String queryParams = "?app_key=" + appKey;
        String response = getRequest("check_app" + queryParams);

        if (response != null && !response.isEmpty()) {
            Map<String, Object> jsonResponse = new ObjectMapper().readValue(response, Map.class);
            return jsonResponse.containsKey("activations_left") ? (int) jsonResponse.get("activations_left") : -1;
        }
        return -1;
    }

    public boolean updateAppKey(String appKey, int numActivations, String accessKey)
            throws IOException, InterruptedException {
        String queryParams = "?app_key=" + appKey + "&num_activations=" + numActivations;
        String response = postRequest("update_app_key" + queryParams, "", accessKey);

        if (response != null && !response.isEmpty()) {
            Map<String, Object> jsonResponse = new ObjectMapper().readValue(response, Map.class);
            return jsonResponse.containsKey("message") && jsonResponse.get("message").toString().equals("App key activations updated successfully");
        }
        return false;
    }

    private String postRequest(String endpoint, String data, String accessToken)
            throws IOException, InterruptedException {
        HttpRequest request = HttpRequest.newBuilder()
                .uri(URI.create(apiUrl + endpoint))
                .header("Content-Type", "application/json")
                .header("Accept", "application/json")
                .header("Authorization", accessToken != null ? "Bearer " + accessToken : "")
                .POST(HttpRequest.BodyPublishers.ofString(data, StandardCharsets.UTF_8))
                .build();

        HttpResponse<String> httpResponse = httpClient.send(request, HttpResponse.BodyHandlers.ofString());

        if (httpResponse.statusCode() == 200) {
            return httpResponse.body();
        }

        return null;
    }

    private String getRequest(String endpoint) throws IOException, InterruptedException {
        HttpRequest request = HttpRequest.newBuilder()
                .uri(URI.create(apiUrl + endpoint))
                .header("Accept", "application/json")
                .GET()
                .build();

        HttpResponse<String> httpResponse = httpClient.send(request, HttpResponse.BodyHandlers.ofString());

        if (httpResponse.statusCode() == 200) {
            return httpResponse.body();
        }

        return null;
    }
}

File: src/C#/ScrewApiClient.cs
using System;
using System.Collections.Generic;
using System.Net.Http;
using System.Net.Http.Headers;
using System.Text;
using System.Threading.Tasks;
using Newtonsoft.Json;

namespace Screw.Api
{
    public class ScrewApiClient
    {
        private readonly HttpClient httpClient;
        private readonly string apiUrl = "https://api.screwltd.com/";

        public ScrewApiClient()
        {
            httpClient = new HttpClient();
            httpClient.BaseAddress = new Uri(apiUrl);
            httpClient.DefaultRequestHeaders.Accept.Clear();
            httpClient.DefaultRequestHeaders.Accept.Add(new MediaTypeWithQualityHeaderValue("application/json"));
        }

        public async Task<string> Authenticate(string login, string password)
        {
            var authData = new Dictionary<string, string>
            {
                { "login", login },
                { "password", password }
            };

            var authJson = JsonConvert.SerializeObject(authData);

            var response = await PostRequest("auth", authJson);
            if (!string.IsNullOrEmpty(response))
            {
                var jsonResponse = JsonConvert.DeserializeObject<Dictionary<string, string>>(response);
                return jsonResponse.ContainsKey("access_key") ? jsonResponse["access_key"] : null;
            }

            return null;
        }

        public async Task<bool> CheckAuthValid(string access_key)
        {
            return !string.IsNullOrEmpty(access_key);
        }

        public async Task<string> CreateAppKey(string accessKey)
        {
            var response = await PostRequest("create_app", string.Empty, accessKey);
            if (!string.IsNullOrEmpty(response))
            {
                var jsonResponse = JsonConvert.DeserializeObject<Dictionary<string, string>>(response);
                return jsonResponse.ContainsKey("app_key") ? jsonResponse["app_key"] : null;
            }
            else
            {
                Console.WriteLine($"Failed to create app key. Response is empty.");
                return null;
            }
        }

        public async Task<bool> AuthenticateApp(string appKey, string accessKey)
        {
            var queryParams = $"?app_key={appKey}";
            var response = await PostRequest("auth_app" + queryParams, string.Empty, accessKey);

            if (!string.IsNullOrEmpty(response))
            {
                var jsonResponse = JsonConvert.DeserializeObject<Dictionary<string, object>>(response);
                return jsonResponse.ContainsKey("result") && (bool)jsonResponse["result"];
            }
            return false;
        }

        public async Task<int> CheckAppKeyActivations(string appKey)
        {
            var queryParams = $"?app_key={appKey}";
            var response = await GetRequest("check_app" + queryParams);

            if (!string.IsNullOrEmpty(response))
            {
                var jsonResponse = JsonConvert.DeserializeObject<Dictionary<string, object>>(response);
                return jsonResponse.ContainsKey("activations_left") ? Convert.ToInt32(jsonResponse["activations_left"]) : -1;
            }
            return -1;
        }

        public async Task<bool> UpdateAppKey(string appKey, int numActivations, string accessKey)
        {
            var queryParams = $"?app_key={appKey}&num_activations={numActivations}";
            var response = await PostRequest("update_app_key" + queryParams, string.Empty, accessKey);

            if (!string.IsNullOrEmpty(response))
            {
                var jsonResponse = JsonConvert.DeserializeObject<Dictionary<string, object>>(response);
                return jsonResponse.ContainsKey("message") && jsonResponse["message"].ToString() == "App key activations updated successfully";
            }
            return false;
        }

        private async Task<string> PostRequest(string endpoint, string data, string accessToken = null)
        {
            httpClient.DefaultRequestHeaders.Authorization = accessToken != null ? new AuthenticationHeaderValue("Bearer", accessToken) : null;
            var content = new StringContent(data, Encoding.UTF8, "application/json");
            var response = await httpClient.PostAsync(endpoint, content);

            if (response.IsSuccessStatusCode)
            {
                return await response.Content.ReadAsStringAsync();
            }

            return null;
        }

        private async Task<string> GetRequest(string endpoint)
        {
            var response = await httpClient.GetAsync(endpoint);

            if (response.IsSuccessStatusCode)
            {
                return await response.Content.ReadAsStringAsync();
            }

            return null;
        }
    }
}
